File: sagrada_app/lib/entities/dice_board.dart
import 'dice.dart';

class DiceBoard {
  DiceBoard()
      : _board = List.generate(rows, (index) => List.filled(columns, null));

  static const int rows = 4;
  static const int columns = 5;

  final List<List<Dice?>> _board;

  void set(int row, int column, Dice dice) {
    assert(row >= 0 && row < rows, 'Row $row is out of bounds');
    assert(column >= 0 && column < columns, 'Column $column is out of bounds');
    _board[row][column] = dice;
  }

  Dice? get(int row, int column) {
    assert(row >= 0 && row < rows, 'Row $row is out of bounds');
    assert(column >= 0 && column < columns, 'Column $column is out of bounds');
    return _board[row][column];
  }
}

File: entities/dice_board.js
export default class DiceBoard {
    static rows = 4;
    static columns = 5;

    constructor() {
        this.dices = new Array(DiceBoard.rows).fill(null)
            .map(() => new Array(DiceBoard.columns).fill(null));
        console.log('DiceBoard created');
    }

    get(row, column) {
        if (row < 0 || row >= this.rows) {
            throw new Error('row must be between 0 and 3');
        }
        if (column < 0 || column >= this.columns) {
            throw new Error('column must be between 0 and 4');
        }
        return this.dices[row][column];
    }

    set(row, column, dice) {
        if (row < 0 || row >= this.rows) {
            throw new Error('row must be between 0 and 3');
        }
        if (column < 0 || column >= this.columns) {
            throw new Error('column must be between 0 and 4');
        }
        this.dices[row][column] = dice;
    }
}
